src/lib/storage/frame_of_reference/frame_of_reference_iterable.hpp

```diff
@@ -0,0 +1,142 @@
+
+ private:
+  template <typename ZsIteratorType>
+  class Iterator : public BaseColumnIterator<Iterator<ZsIteratorType>, ColumnIteratorValue<T>> {
+   public:
+    using ReferenceFrameIterator = typename pmr_vector<T>::const_iterator;
+    using OffsetValueIterator = ZsIteratorType;
+    using NullValueIterator = typename pmr_vector<bool>::const_iterator;
+
+   public:
+    // Begin Iterator
+    explicit Iterator(ReferenceFrameIterator block_minimum_it, OffsetValueIterator offset_value_it,
+                      NullValueIterator null_value_it)
+        : _block_minimum_it{block_minimum_it},
+          _offset_value_it{offset_value_it},
+          _null_value_it{null_value_it},
+          _index_within_frame{0u},
+          _chunk_offset{0u} {}
+
+    // End iterator
+    explicit Iterator(OffsetValueIterator offset_value_it) : Iterator{{}, offset_value_it, {}} {}
+
+   private:
+    friend class boost::iterator_core_access;  // grants the boost::iterator_facade access to the private interface
+
+    void increment() {
+      ++_offset_value_it;
+      ++_null_value_it;
+      ++_index_within_frame;
+      ++_chunk_offset;
+
+      if (_index_within_frame >= FrameOfReferenceColumn<T>::block_size) {
+        _index_within_frame = 0u;
+        ++_block_minimum_it;
+      }
+    }
+
+    bool equal(const Iterator& other) const { return _offset_value_it == other._offset_value_it; }
+
+    ColumnIteratorValue<T> dereference() const {
+      const auto value = static_cast<T>(*_offset_value_it) + *_block_minimum_it;
+      return ColumnIteratorValue<T>{value, *_null_value_it, _chunk_offset};
+    }
+
+   private:
+    ReferenceFrameIterator _block_minimum_it;
+    OffsetValueIterator _offset_value_it;
+    NullValueIterator _null_value_it;
+    size_t _index_within_frame;
+    ChunkOffset _chunk_offset;
+  };
+
+  template <typename ZsDecoderType>
+  class PointAccessIterator
+      : public BasePointAccessColumnIterator<PointAccessIterator<ZsDecoderType>, ColumnIteratorValue<T>> {
+   public:
+    // Begin Iterator
+    PointAccessIterator(const pmr_vector<T>* block_minima, const pmr_vector<bool>* null_values,
+                        ZsDecoderType* attribute_decoder, ChunkOffsetsIterator chunk_offsets_it)
+        : BasePointAccessColumnIterator<PointAccessIterator<ZsDecoderType>, ColumnIteratorValue<T>>{chunk_offsets_it},
+          _block_minima{block_minima},
+          _null_values{null_values},
+          _attribute_decoder{attribute_decoder} {}
+
+    // End Iterator
+    explicit PointAccessIterator(ChunkOffsetsIterator chunk_offsets_it)
+        : PointAccessIterator{nullptr, nullptr, nullptr, chunk_offsets_it} {}
+
+   private:
+    friend class boost::iterator_core_access;  // grants the boost::iterator_facade access to the private interface
+
+    ColumnIteratorValue<T> dereference() const {
+      const auto& chunk_offsets = this->chunk_offsets();
+
+      if (chunk_offsets.into_referenced == INVALID_CHUNK_OFFSET)
```

Comment:
Comment? What does `INVALID_CHUNK_OFFSET` represent here?

Reply:
After Moritz’s changes, this can probably be removed. It represents NULL

Reply:
Which changes do you mean?

Reply:
Some bug fixes from the const table layout PR. I have removed this line.